### ISSUE TITLE
Local start does not work for Windows

### ISSUE THREAD

**Issue by sanstechnologies** [Azure-Samples/azure-search-openai-demo#762]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>
run app/start.ps1 on windows computer
### Any log messages given by the failure
> uvloop is not supported for Windows

### Expected/desired behavior
> deploy the app to localhost:50505

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
Windows 11

### azd version?
> run `azd version` and copy paste here.
1.4.1
### Versions
>

### Mention any other details that might be useful
If the uvloop and uvicorn are commented out from the backend/requirements.txt the application will come up.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
What Python version are you using? For 3.12, there are various dependency issues, but it should work in 3.11 - 3.9. I believe uvicorn is supported on Windows (it's mentioned on https://www.uvicorn.org/ and I haven't heard other reports).

**Comment by sunnygambit1989** [comment]
@pamelafox facing same error. Earlier before oct 2 commit . It was working fine.

**Comment by leongj** [comment]
Hey @pamelafox - just hit this with one of our customers -- they are using Python 3.12.

```
Collecting uvicorn[standard]==0.23.2 (from -r requirements.txt (line 299))
  Obtaining dependency information for uvicorn[standard]==0.23.2 from https://files.pythonhosted.org/packages/79/96/b0882a1c3f7ef3dd86879e041212ae5b62b4bd352320889231cc735a8e8f/uvicorn-0.23.2-py3-none-any.whl.metadata
  Downloading uvicorn-0.23.2-py3-none-any.whl.metadata (6.2 kB)
Collecting uvloop==0.17.0 (from -r requirements.txt (line 301))
  Downloading uvloop-0.17.0.tar.gz (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 9.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
 
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "C:\Users\carolana\OneDrive - Ventia\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\carolana\OneDrive - Ventia\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main  
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\carolana\OneDrive - Ventia\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\carolana\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\carolana\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "C:\Users\carolana\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\carolana\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 8, in <module>
      RuntimeError: uvloop does not support Windows at the moment
      [end of output]
 
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
 
× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
 
note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by leongj** [comment]
Hey @pamelafox -- here's an example of the error (on Python 3.12)

```
Collecting uvicorn[standard]==0.23.2 (from -r requirements.txt (line 299))
  Obtaining dependency information for uvicorn[standard]==0.23.2 from https://files.pythonhosted.org/packages/79/96/b0882a1c3f7ef3dd86879e041212ae5b62b4bd352320889231cc735a8e8f/uvicorn-0.23.2-py3-none-any.whl.metadata
  Downloading uvicorn-0.23.2-py3-none-any.whl.metadata (6.2 kB)
Collecting uvloop==0.17.0 (from -r requirements.txt (line 301))
  Downloading uvloop-0.17.0.tar.gz (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 9.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
 
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "C:\Users\auser\OneDrive\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\auser\OneDrive\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main  
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\auser\OneDrive\Documents\repos\demo\azure-search-openai-demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\auser\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\auser\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "C:\Users\auser\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\auser\AppData\Local\Temp\pip-build-env-_65ddcu7\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 8, in <module>
      RuntimeError: uvloop does not support Windows at the moment
      [end of output]
 
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
 
× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
 
note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by sunnygambit1989** [comment]
@pamelafox  
equirement already satisfied: uvicorn[standard]==0.23.2 in d:\azure-open_ai_demo\app\backend\backend_env\lib\site-packages (from -r requirements.txt (line 299)) (0.23.2)
Collecting uvloop==0.17.0 (from -r requirements.txt (line 301))
  Using cached uvloop-0.17.0.tar.gz (2.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "D:\azure-open_ai_demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "D:\azure-open_ai_demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "D:\azure-open_ai_demo\app\backend\backend_env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Lenovo\AppData\Local\Temp\pip-build-env-epo79xzi\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Lenovo\AppData\Local\Temp\pip-build-env-epo79xzi\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "C:\Users\Lenovo\AppData\Local\Temp\pip-build-env-epo79xzi\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\Lenovo\AppData\Local\Temp\pip-build-env-epo79xzi\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 8, in <module>
      RuntimeError: uvloop does not support Windows at the moment
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.2.1 -> 23.3
[notice] To update, run: D:\azure-open_ai_demo\app\backend\backend_env\scripts\python.exe -m pip install --upgrade pip

Restoring frontend npm packages

**Comment by pamelafox** [comment]
Thanks! I'll send out a fix.
